### DISCUSSION TITLE
mini.ai: troubles defining number with whitespace

### DISCUSSION THREAD

**Discussion by simonmandlik** [nvim-mini/mini.nvim#971] [Q&A] [answered]
Hello,

I'm trying to define "number" textobject. Basically, I'm combining these two examples from the docs:

- "Number": `{ '%f[%d]%d+' }` 
- "Imitate word ignoring digits and punctuation (supports only Latin alphabet)": `{ '()()%f[%w]%w+()[ \t]*()' }`

Just replacing `%w` with `%d` I arrived at

```lua
custom_textobjects = {
  U = { "()()%f[%w]%w+()[ \t]*()" },
  S = { "()()%f[%d]%d+()[ \t]*()" }
}
```

Testing on `1234    `, `vaU` correctly selects the whole line, whereas `vaS` only selects the first space. Where is the mistake please?

**Comment by echasnovski** [comment, marked as answer]
> Just replacing `%w` with `%d` I arrived at
> 
> ```lua
> custom_textobjects = {
>   U = { "()()%f[%w]%w+()[ \t]*()" },
>   S = { "()()%f[%d]%d+()[ \t]*()" }
> }
> ```
> 
> Testing on `1234 `, `vaU` correctly selects the whole line, whereas `vaS` only selects the first space. Where is the mistake please?

For me both `vaU` and `vaS` select `1234` *and* the whole whitespace to the right. This is the whole 'mini.ai' setup:
```lua
local ai = require('mini.ai')
ai.setup({
  custom_textobjects = {
    U = { '()()%f[%w]%w+()[ \t]*()' },
    S = { '()()%f[%d]%d+()[ \t]*()' },
  },
})
```

Make sure that `S` is not being overridden by something else. Output of `:=MiniAi.config.custom_textobjects` should show exactly what you'd expect.

**Reply by simonmandlik** [reply]
Yeah, turns out I had a different remap elsewhere, sorry 🫣
